src/Pages/EditClassForm.jsx

```diff
@@ -0,0 +1,220 @@
+    const [gender, setGender] = useState(classToEdit.trainer.gender);
+
+    // useEffect(() => {
+    //     console.log("Inside class to edit: ", classToEdit)
+    //     if (classToEdit) {
+    //         setClassName(classToEdit.className)
+    //         setClassType(classToEdit.classType)
+    //         setClassStatus(classToEdit.classStatus)
+    //         setClassStartDate(classToEdit.setClassStartDate)
+    //         setDuration(classToEdit.duration)
+    //         setFullName(classToEdit.fullName)
+    //         setSpeciality(classToEdit.speciality)
+    //         setQualification(classToEdit.qualification)
+    //         setGender(classToEdit.gender)
+    //     }
+    // }, [classToEdit])
+
+    const handleClassName = (event) => {
+        event.preventDefault()
+        setClassName(event.target.value)
+    }
+
+    const handleClassType = (event) => {
+        event.preventDefault()
+        setClassType(event.target.value)
+    }
+
+    const handleClassStatus = (event) => {
+        event.preventDefault()
+        setClassStatus(event.target.value)
+    }
+
+    const handleClassStartDate = (event) => {
+        event.preventDefault()
+        setClassStartDate(event.target.value)
+    }
+
+    const handleDuration = (event) => {
+        event.preventDefault()
+        setDuration(event.target.value)
+    }
+    const handleFullName = (event) => {
+        event.preventDefault()
+        setFullName(event.target.value)
+    }
+
+    const handleSpeciality = (event) => {
+        event.preventDefault()
+        setSpeciality(event.target.value)
+    }
+
+    const handleQualification = (event) => {
+        event.preventDefault()
+        setQualification(event.target.value)
+    }
+
+    const handleGender = (event) => {
+        event.preventDefault()
+        setGender(event.target.value)
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+
+        const date = new Date(classStartDate)
+
+        const classToUpdate = {
+            className,
+            classType,
+            classStatus,
+            classStartDate: date,
+            duration,
+            trainer: {
+                fullName,
+                speciality,
+                qualification,
+                gender,
+            }
+        }
+
+        const fetchOptions = {
+            method: "PUT",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(classToUpdate)
+        };
+
+        fetch(`${API_URL}/classes/${classId}`, fetchOptions)
+            .then((res) => res.json())
+            .then((classToUpdate) => {
+                console.log("Updated class", classToUpdate);
+
+                const updatedClass = classes.map((classe) => {
+                    if (classToUpdate.id === classe.id) {
+                        return classToUpdate;
+                    } else {
+                        return classe
+                    }
```

Comment:
You don't need to change this, but this if statement is a perfect opportunity to use a ternary (short-hand if) to bring this code block down to a single line:

```
const updatedClass = classes.map(classe => classToUpdate.id === classe.id ? classToUpdate : classe);
```